MoreLinq.Test/TakeLastTest.cs

```diff
@@ -73,11 +57,29 @@ public void TakeLastDisposesSequenceEnumerator()
             {
                 seq.TakeLast(1).Consume();
             }
+        }
 
-            using (var seq = new[]{ 1,2,3 }.AsTestingSequence())
-            {
-                seq.TakeLast(1).Consume();
-            }
+        // ICollection<T> Optimization Tests
```

Comment:
> These tests don't test for optimisation so the comment is misleading. They only test that the behaviour of TakeLast does not change when passed a collection, which is why removing the optimisation in TakeLast does not fail them.

I see what you and @fsateler meant now, thanks for clarify.

> It seems futile to test that TakeLast is optimised for collections. The best one can do is to assert that the Count property was used, which doesn't seem like a very useful assurance.

I agree, it doesnt seem useful.

> My main problem with 4ce5833 was that it seemed to abuse local functions when a simpler and more classic approach to have a (locally) reusable assertion function would be sufficient

I agree. probrably I've got too excited about local functions that I didn't use it appropriately.
I will replace the current tests by that classic approach you show.

> Also agree with @fsateler that the tests should fail if the optimisation is removed and yet they don't.

how can I make a test that fail if the optimisation is removed?